### DISCUSSION TITLE
How can I analyze a message sent on a chat?

### DISCUSSION THREAD

**Discussion by pepo-ec** [def-studio/telegraph#119] [Q&A]
I created a chat with my bot and using "Telegraph" I can send messages but I still don't understand how I can read the messages that have been written in the chat. Can you please explain to me (with an example) how handlers and their functions are used?

**Comment by fabio-ivona** [comment]
Hi! You can achieve it by using both:

- webooks: https://def-studio.github.io/telegraph/webhooks/overview

- manual polling: https://def-studio.github.io/telegraph/webhooks/manual-polling

**Reply by pepo-ec** [reply]
I am using "Laravel": Should I create a route to execute a specific procedure/handler?. How do I call the handler who will respond when Telegram receives a message?

**Reply by pepo-ec** [reply]
I have changed config/telegraph.php file:

`    'webhook_handler' => App\Http\Webhooks\MyWebhookHandler::class,`

And my MyWebhookHandler.php file is:

```
<?php

namespace App\Http\Webhooks;

use DefStudio\Telegraph\Handlers\WebhookHandler;
use DefStudio\Telegraph\Models\TelegraphChat;
use DefStudio\Telegraph\Models\TelegraphBot;
use Illuminate\Http\Request;
use Illuminate\Http\Response;

class MyWebhookHandler extends \DefStudio\Telegraph\Handlers\WebhookHandler
{
    public function handle(Request $request, TelegraphBot $bot): void
    {
        $chat = TelegraphChat::find(1);
        $chat->message('mensaje de respuesta')->send();
    }
}
```

**Is it the right way to address messages?**

**Reply by fabio-ivona** [reply]
You are on the right way, remember to allow your bot to listen for all messages

**Reply by ehs4nm** [reply]
Hi. I am using the same way. set web-hook. created the same MyWebhookHandler class. But i cant receive anything from my bot. I am chatting with the bot and created the chat and bot. I can send message to this chat but no updates from it?

Asking for a favor: Is it possible to create a tutorial explaining how to get started with basics? setting any route that should be or creating a simple callback, creating a command in the bot.

Or a simple demo project which could be a sample of some basics.

Thanks for this great package it seems to make lots of work done.

**Reply by fabio-ivona** [reply]
@ehs4nm Telegraph has evolved since this conversation started, we have just updated the documentation

now you can handle received messages by overriding `WebhookHandler::handleChatMessage()` method:

```php
class MyWebhookHandler extends WebhookHandler
{
    protected function handleChatMessage(Stringable $text): void
    {
        $this->chat->html("Received: $text")->send();
    }
}
```


> Asking for a favor: Is it possible to create a tutorial explaining how to get started with basics? setting any route that should be or creating a simple callback, creating a command in the bot.
> 
> Or a simple demo project which could be a sample of some basics.

this is a great idea, I will work on something in the next few days

**Comment by ah-rahimi** [comment]
I want to read file in chat with webhook
how about it?
@fabio-ivona

**Reply by fabio-ivona** [reply]
@pepo-ec and @ahr1384 , you should use this:

```php
class MyWebhookHandler extends \DefStudio\Telegraph\Handlers\WebhookHandler
{

    protected function handleChatMessage(Stringable $text): void
    {
       $this->message->text();
       $this->message->photos();
       $this->message->video();
    }
 
}


```
